src/Board.tsx

```diff
@@ -1,34 +1,21 @@
-import * as React from 'react';
+import React, { FunctionComponent } from 'react';
 import { Square } from './Square';
 
 type Props = {
   squares: string[];
+  clickHandler: Function;
 };
 
-export class Board extends React.Component<Props> {
-  public constructor(props: Props) {
-    super(props);
-  }
+export const Board: FunctionComponent<Props> = (props) => {
+  const renderSquare = function(mark: string, pos: number): JSX.Element {
+    return <Square value={mark} clickHandler={() => props.clickHandler(pos)} />;
+  };
 
-  protected renderSquare(mark: string): JSX.Element {
-    return <Square value={mark} />;
-  }
+  const board = Array.from(Array(9).keys()).map(
+    (i: number) => <div className="octothorp" key={i}>{renderSquare(props.squares[i], i)}</div>
```

Comment:
usually have (i: number) on same line as map

Reply:
The line gets pretty long that way though... 🤔

Reply:
can break it up into multiple lines setting board array keys, etc

Reply:
you can use brackets to break it up:

```.map((i: number) => {
  return <div className="octothorblah" />
}```

Reply:
Just to clarify, this would be preferable?

```ts
  const board = Array.from(Array(9).keys())
    .map((i: number) => <div className="octothorp" key={i}>{renderSquare(props.squares[i], i)}</div>);
```

Excluding the form as it currently exists in the commit, the below version would actually be more readable to me since it focuses attention on the JSX over the JS nonsense related to generating `[0, 1, 2, 3, 4, 5, 6, 7, 8]`. Of course this sacrifices the single-line closure form but achieves your goal of getting `.map(` and `(i: number)` on the same line and further reduces line length as a bonus.

```ts
  const board = Array.from(Array(9).keys()).map((i: number) => {
    return (
      <div className="octothorp" key={i}>{renderSquare(props.squares[i], i)}</div>
    );
  });
```

I'd love to hear your reasoning for preferring one over another.

Reply:
yeah the below version is what I was suggesting exactly

Reply:
```
const board = Array.from(Array(9).keys()).map((i: number) => {
    return (
      <div className="octothorp" key={i}>{renderSquare(props.squares[i], i)}</div>
    );
  });
```

Reply:
Our replies crossed. 😆